backend/libs/watcher-operations.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { DocumentClient } from 'aws-sdk/clients/dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import { Room, VideoSyncStatus } from '../../extension/src/types';
+import {
+    MediaEventType,
+    PlayerEvent,
+} from './../../extension/src/contentscript/player';
+import { UserVideoStatus } from './../../extension/src/types';
+import { createRoom, joinExistingRoom, updateRoom } from './room-operations';
+import { updateWatcherVideoStatus } from './watcher-operations';
+
+const isTest = process.env.JEST_WORKER_ID;
+const config = {
+    convertEmptyValues: true,
+    ...(isTest && {
+        endpoint: 'localhost:8000',
+        sslEnabled: false,
+        region: 'local-env',
+    }),
+};
+const ddb = new DocumentClient(config);
+const tableName = 'watch-with-me-dev-RoomTable';
+
+let roomId: string;
+let room: Room | undefined = undefined;
+let playerEvent: PlayerEvent;
+let ddbRoom: DocumentClient.AttributeMap | undefined = undefined;
+
+const getDDBRoom = async () => {
+    const { Item: item } = await ddb
+        .get({
+            TableName: tableName,
+            Key: { roomId },
+        })
+        .promise();
+    if (!item) {
+        throw new Error('Error retrieving DDB room');
+    }
+    ddbRoom = item;
```

Comment:
I'd rather we return the room here and that the place where this function is called handle the side effect

Reply:
and we should pass the roomId as a param,.,

Reply:
The problem is the decomposition between before() and it() blocks. Often the getDDBRoom call is made in a before block and therefore we need to use a global variable that can be asserted in the test.

Also, I'm actually about to push some changes that will convert all beforeEach into beforeAll for more consistent and efficient testing

Reply:
You could still have a `let ddbRoom;` in the `describe` scope, make the `getDDBRoom` function return the `item`, and then do `ddbRoom = getDDBRoom()` in the `before`

Reply:
Tell me what you think of the new tests